### ISSUE TITLE
Begin validation steps

### ISSUE THREAD

**Issue by qjhart** [ucd-cws/calvin-network-data#4]
I guess this will be something like `grunt validate`.  And we can have sub-tasks like `grunt validate-links` ?  First one should test that the filesystem links and nodes corresponds to the origins and destinations in the node definitions.  Should have a reporting and a change the node to match function.  Also can specify the node(s) and or link(s) on cmd line.  If a link is missing a corresponding node, I guess we just report it, and don't delete? 

```
# -n = dry-run just report(s) changes
grunt validate-links -n
C117 {'origins':['N2','N3'],'terminals:['R4']} => {'origins':['N2','S5'],'terminals':['R4]'}
C117_N3 {origin:'C117', terminal:null} 
# Just node C117 
# -l means validate all links too.
grunt validate-links -l C117
C117_N3 {origin:'C117', terminal:null} 
```

**Comment by qjhart** [comment]
I'm thinking mostly fur us to come up with a good framework to plug code into.  As per our discussion, maybe this should be moved the the calvin-network-editor, and leave the data clean.

**Comment by jrmerz** [comment]
@qjhart we closing this?

**Comment by qjhart** [comment]
No, this is part of the ts4 validation as well, which I started but didn't finish.  On top of the json-schema validation needs a layer of consistancy checking as described above.

**Comment by qjhart** [comment]
Another tool to consider with validation is the Terraformer toolkit, http://terraformer.io/core/
